tests/test_gates.py

```diff
@@ -378,6 +378,7 @@ def circuit():
         qml.PauliY,
         qml.PauliZ,
         qml.Hadamard,
+        qml.SX,
```

Comment:
This new implementation will compute faster the SX gate because it uses compilated language (C++) compared with the implementation in Python which is an interpretative language.

Additionally, the Pennylane Lightning code has different layers of optimization such as

* Vectorization with AVX2 or AVX512
* Parallelization in the computing of the gates with OpenMP
* Use contsexpr, when possible, to evaluate at compile time
* The definition of the gate with lambda functions
* Bitwise operations to compute the indices of the state vector

All these options contribute to increasing the performance not only of this gate but also of all the gates implemented.